_posts/en/newsletters/2023-10-11-newsletter.md

```diff
@@ -0,0 +1,189 @@
+      wrong type of transaction ID (but consistently) in the first place."
+  a2link="https://bitcoincore.reviews/28107#l-67"
+
+  q3="Conceptually, when writing new code that requires referring to
+      transactions, when should you use `txid` and when should you use
+      `wtxid`? Can you point to any examples in the code where using one
+      instead of the other could be very bad?"
+  a3="In general, use of `wtxid` is preferred since it commits to the
+      entire transaction. An important exception is the `prevout`
+      reference from each input to the output (UTXO) it's spending.
+      It must be a `txid` to retain compatibility with pre-segwit nodes.
+      (Note that it's not possible for only upgraded segwit nodes to use
+      `wtxid` here, because this would change the `txid` of the spending
+      transaction.)
+      An example of where it's important to use one and not the other is
+      given [here][wtxid example]."
+  a3link="https://bitcoincore.reviews/28107#l-85"
+
+  q4="In which concrete way(s) could using `transaction_identifier` instead
+      of `uint256` help find existing bugs or prevent the introduction of
+      new ones? On the other hand, could this change introduce new bugs?"
+  a4="Without this PR, a function that takes a `uint256` argument (such as
+      a block ID hash) could be passed a `txid`.
+      With this PR, this causes a compile-time error."
+  a4link="https://bitcoincore.reviews/28107#l-128"
+
+  q5="The [`GenTxid`][GenTxid] class already exists. How does it
+      already enforce type correctness, and how is it different from the
+      approach in this PR?"
+  a5="This class includes a hash and a flag indicating if the hash is a
+      `wtxid` or a `txid`, so it's still a single type rather than two
+      distinct types. This allows type checking, but it must be explicitly
+      programmed, and, more importantly, can only be done at run-time,
+      not compile-time."
+  a5link="https://bitcoincore.reviews/28107#l-161"
+
+  q6="How is `transaction_identifier` able to subclass `uint256`, given
+      that, in C++, integers are types and not classes?"
+  a6="Because `uint256` is itself a class, rather than a built-in type.
+      (The largest built-in C++ integer type is 64 bits.)"
+  a6link="https://bitcoincore.reviews/28107#l-194"
+
+  q7="Does a `uint256` behaves otherwise the same as, for example, a
+      `uint64_t?`"
+  a7="No, arithmetic operations aren't permitted on `uint256` because
+      they don't make sense for hashes (which is the main use of `uint256`).
+      The name is misleading; it's really just a blob of 256 bits.
+      A separate `arith_uint256 ` allows arithmetic (used, for example,
+      in PoW calculations)."
+  a7link="https://bitcoincore.reviews/28107#l-203"
+
+  q8="Why does `transaction_identifier` subclass `uint256` instead of
+      being a completely new type?"
+  a8="It allows us to use explicit and implicit conversions to leave
```

Comment:
Maybe? 
```suggestion
  a8="It allows us to use explicit and implicit conversions and to leave
```

Reply:
I don't think we're making two separate points here; the conversions allow us to leave some code unchanged. UUIC, it's better as is.